### PR TITLE
Remove non-gmp integers implementation

### PR DESCRIPTION
**Pull request by markuspf**
This removes the GAP non-gmp integer implementation.

I left the `integer.h` file in for backwards compatibility. This is not meant for merging as-is as I figure there's something to be done in the build system with the `--with-gmp` flag. Also I need to rearrange and squish some commits.

If @fingolfin could hint me at the correct places to tweak the build system I am happy to do this.

### CODE REVIEW

**Comment by markuspf** [comment]
Also if anyone knows any other places we need to purge, then speak up.

File: src/integer.h [hunk 1]
```diff
@@ -17,235 +17,8 @@
...
-*/
-#ifdef SYS_IS_64_BIT
-typedef UInt4           TypDigit;
-#define INTEGER_UNIT_SIZE 4
-#define INTEGER_ALLOCATION_SIZE 16
 #else
-typedef UInt2           TypDigit;
-#define INTEGER_UNIT_SIZE 2
-#define INTEGER_ALLOCATION_SIZE 8
+ #warning "GAP integer code has been removed."
```
Comment (ChrisJefferson) [inline src/integer.h]: Minor pick, this may as well be #error, else users will (I think) end up with a link-time error.

**Comment by ChrisJefferson** [comment]
`vecgf2.c` includes a function which has a different implementation with, and without, `USE_GMP`.

**Comment by markuspf** [comment]
Packages `io` and `fr` seem to be referring to `USE_GMP` as well. I removed `USE_GMP` everywhere except `integer.h`

**Comment by fingolfin** [comment]
@markuspf I am not quite sure what exactly you feel should be changed in the build system? Perhaps you want an error if no GMP is found or something like that? If you let me know what you think needs to be tweaked, I can certainly provide suggestions. Until then, what you did so far already looks quite comprehensive.

**Comment by markuspf** [comment]
@fingolfin I think mainly removing the --with-gmp=no option and possibly removing or deprecating `USE_GMP`?

**Comment by fingolfin** [comment]
I don't think we should remove `USE_GMP` anytime soon, if ever -- several packages check for it (and may error out if it is missing). It doesn't cause us harm, either, so I'd just keep it (perhaps with a comment next to it explaining why it's there). Of course we can try to get package authors to stop using USE_GMP, and depending on GAP 4.9, but that will take time (months at the very least).

As to removing --with-gmp=no: I don't think we can easily remove it. And `autoconf` also allows you to write `--without-gmp`. Instead, let's simply change the check for `with_gmp` to print a helpful error if `with_gmp=no` holds.

**Comment by markuspf** [comment]
Right, we'll also have to remove the mentions of "or compile GAP without GMP" because that's not possible anymore then.

**Comment by markuspf** [comment]
Now for some history revision...

**Comment by markuspf** [comment]
Just in case: I think this is ready for merge. I also tested building packages with this PR applied and so far there are no problems.

**Comment by alex-konovalov** [comment]
The kind of an error which I am happy to report
```
configure: error: GMP is required to build GAP. You can specify "--with-gmp=yes" to build the default version of GMP shipped with GAP,
                 "--with-gmp=system" to use the default system search paths "/usr" and "/usr/local", or a path to a precompiled
                 version of GMP. The option "--with-gmp=no" to compile GAP without GMP is no longer supported.
Build step 'Execute shell' marked build as failure
```
After tonight's tests, I will reconfigure nightly tests in the master branch to use only default "--with-gmp=yes".

**Pull request merged**